composer.json

```diff
@@ -39,7 +39,9 @@
     "require": {
         "php": "^5.5.0|^7.0",
         "payum/iso4217": "~1.0",
-        "guzzlehttp/guzzle": "~6.0",
+        "php-http/client-implementation": "^1.0",
+        "php-http/discovery": "^0.8",
+        "guzzlehttp/psr7": "^1.1",
```

Comment:
Payum. Every time you [create a Request](https://github.com/Payum/Payum/blob/master/src/Payum/Skeleton/Api.php#L42) you use the Guzzle implementation. 
I could use Httplug's MessageFactory interface to decouple myself from any PSR7 implementation.